1022 moving moBi excel import logic to core

georgeDaskalakis opened the pull request:
fixes #1022

msevestre commented on src/OSPSuite.Infrastructure.Import/Core/DataSourceFileReaders/ExcelReader.cs:
```diff
@@ -84,6 +84,21 @@ public bool MoveToNextRow()
+
+      private bool isSheetEmpty(ISheet sheet)
+      {
+
```
can you run resharper please to style this file?

georgeDaskalakis replied:
done. (it removed a few empty lines mainly)

msevestre commented on src/OSPSuite.Infrastructure.Import/Core/DataSourceFileReaders/ExcelReader.cs:
```diff
@@ -84,6 +84,21 @@ public bool MoveToNextRow()
          return true;
       }
 
+      public IEnumerable<string> RetrieveExcelSheets(bool excludeEmptySheets)
```
should that have a default of true (excludeEmptySheetS)

georgeDaskalakis replied:
I agree, it is better with a default value. Done.

msevestre commented on src/OSPSuite.Infrastructure.Import/Extensions/ExcelReadingExtensions.cs:
```diff
@@ -0,0 +1,25 @@
+         var i = 0; //IS THIS ZERO ACTUALLY RIGHT????
+         foreach (var cellContent in rowValues)
+         {
+            dataRow[i] = cellContent;
```
what is this? lol . You are always overriding the same entry

georgeDaskalakis replied:
corrected

msevestre commented on src/OSPSuite.Infrastructure.Import/Services/IImportFromExcelTask.cs:
```diff
@@ -0,0 +1,111 @@
+         }
+         catch
+         {
+            return Enumerable.Empty<string>().ToList();
```
Array.empty

georgeDaskalakis replied:
corrected

msevestre commented on src/OSPSuite.Infrastructure.Import/Services/IImportFromExcelTask.cs:
```diff
@@ -0,0 +1,111 @@
+
+   public class ImportFromExcelTask : IImportFromExcelTask
+   {
+      public IReadOnlyList<string> RetrieveExcelSheets(string fileName, bool excludeEmptySheets = false)
```
why is default false? Does not true make more sense?

georgeDaskalakis replied:
I have kept this exactly the way it was in MoBi. My understanding is that if you just want the list of names of all the sheets, it is the default that you get even the empty ones, since they exist as sheets.

msevestre reviewed (changes requested):
all good except this AddRowToDataTable which looks very suspicious to me
And please run the formatted:)

georgeDaskalakis commented:
> all good except this AddRowToDataTable which looks very suspicious to me
> And please run the formatted:)

but you mean just the format here? The formatter did not change anything. But I do think I will rename the class to DataTableExtension, it makes more sense.

msevestre commented:
no I mean the code in AddRowToDataTable  seems wrong. You have an index I that does not change in the loop. WTF?

georgeDaskalakis commented:
Yeap, about that you are right, I just corrected it. But since it was also in a different comment I got confused about what you meant. I will also add a unit test for the reading - the existing unit tests that were in MoBi and I moved to Core are not testing if any of the values read are correct (possibly also did not need to since SmartXLS seemed to have a method to directly return a DataTable).

msevestre commented:
@georgeDaskalakis Did you push? Still looks off to me

msevestre commented:
@georgeDaskalakis Can you push your changes when you have time. I can't merge the PR with some broken code sorry

georgeDaskalakis commented:
Sorry, I pushed just now because I needed to clarify one thing and finish the unit test. I ended up renaming the extension class to ImporterDataTableExtensions. There exists also a DataTableExtensions class in OSPSuite.Utility, which does not have this add List<string> as row functionality. Sill, since this is too specific, I would not like to extend the Utilities.
